curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -1333,8 +1457,11 @@ void DataCache::Write(uint64_t chunkPos, uint64_t len, const char *data,
                            (*iter)->GetData() +
                                (chunkPos + len - (*iter)->GetChunkPos()),
                            totalSize - len);
+                    chunkCacheManager_->rwLockWrite_.WRLock();
                     Swap(newDatabuf, totalSize);
                     chunkPos_ = chunkPos;
+                    chunkCacheManager_->UpdateWrteCacheMap(oldChunkPos);
```

Comment:
lock can be placed in function UpdateWrteCacheMap

Reply:
it must passed A lot of arguments to UpdateWrteCacheMap